src/identity/identity.service.spec.ts

```diff
@@ -0,0 +1,181 @@
+
+      expect(spies.storage.getPublicKey.mock.calls.length).toBe(1);
+      expect(spies.verificationMethod.getMethodsFor.mock.calls.length).toBe(1);
+
+      expect(did).toEqual({
+        '@context': 'https://www.w3.org/ns/did/v1',
+        id: 'did:lto:sender',
+        verificationMethod: [{
+          id: 'did:lto:sender#key',
+          type: 'Ed25519VerificationKey2018',
+          controller: 'did:lto:sender',
+          publicKeyBase58: 'publicKey',
+          blockchainAccountId: 'sender@lto:chainId',
+        }, {
+          id: 'did:lto:recipient#key',
+          type: 'Ed25519VerificationKey2018',
+          controller: 'did:lto:recipient',
+          publicKeyBase58: 'publicKey',
+          blockchainAccountId: 'recipient@lto:chainId',
+        }],
+        authentication: [
+          'did:lto:sender#key',
+          'did:lto:recipient#key',
+        ],
+        assertionMethod: [
+          'did:lto:sender#key',
+          'did:lto:recipient#key',
+        ],
+        keyAgreement: [
+          'did:lto:recipient#key',
+        ],
+        capabilityInvocation: [
+          'did:lto:sender#key',
+        ]
+      });
+    });
+
+    test('should resolve the identity with MULTIPLE additional verification methods', async () => {
+      const spies = spy();
+
+      spies.verificationMethod.getMethodsFor = jest.spyOn(verificationMethodService, 'getMethodsFor').mockImplementation(async (address: string) => {
+        const relationships = 0x0107; // authentication, assertion, key agreement
+        const secondRelationships = 0x0113; // authentication, assertion, capabilityDelegation
+
+        return [new VerificationMethod(relationships, address, 'recipient'), new VerificationMethod(secondRelationships, address, 'secondRecipient')];
+      });
+
+      const did = await identityService.resolve('sender');
+
+      expect(spies.storage.getPublicKey.mock.calls.length).toBe(1);
+      expect(spies.verificationMethod.getMethodsFor.mock.calls.length).toBe(1);
+
+      expect(did).toEqual({
+        '@context': 'https://www.w3.org/ns/did/v1',
+        id: 'did:lto:sender',
+        verificationMethod: [{
+          id: 'did:lto:sender#key',
+          type: 'Ed25519VerificationKey2018',
+          controller: 'did:lto:sender',
+          publicKeyBase58: 'publicKey',
+          blockchainAccountId: 'sender@lto:chainId',
+        }, {
+          id: 'did:lto:recipient#key',
+          type: 'Ed25519VerificationKey2018',
+          controller: 'did:lto:recipient',
+          publicKeyBase58: 'publicKey',
+          blockchainAccountId: 'recipient@lto:chainId',
+        }, {
+          id: 'did:lto:secondRecipient#key',
+          type: 'Ed25519VerificationKey2018',
+          controller: 'did:lto:secondRecipient',
+          publicKeyBase58: 'publicKey',
+          blockchainAccountId: 'secondRecipient@lto:chainId',
+        }],
+        authentication: [
+          'did:lto:sender#key',
```

Comment:
If there are one or more verification methods (through associations) for the identity, don't add `'did:lto:sender#key'` as `authentication`, `assertionMethod`, and `capabilityInvocation`.

Reply:
done